src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -51575,6 +51575,62 @@ static void F()
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOfTargetDelegate, "(I<object?> o) => { }").WithArguments("o", "lambda expression", "D<I<object>>").WithLocation(10, 26));
         }
 
+        [Fact, WorkItem(40561, "https://github.com/dotnet/roslyn/issues/40561")]
+        public void ReturnLambda_InsideConditionalExpr()
+        {
+            var source = @"
+using System;
+
+class C
+{
+    static void M0(string s) { }
+
+    static Action? M1(string? s)
+        => s != null
+            ? () => M0(s)
+            : (Action?)null;
+
+    static Action? M2(string? s)
+        => s != null
+            ? (Action)(() => M0(s))
+            : null;
+
+    static Action? M3(string? s)
+        => s != null
+            ? () => { M0(s); s = null; }
+            : (Action?)null;
+
+    static Action? M4(string? s)
+    {
+        return s != null
+            ? local(() => M0(s), s = null)
+            : (Action?)null;
+
+        Action local(Action a1, string? s)
+        {
+            return a1;
+        }
+    }
+
+    static Action? M5(string? s)
+    {
+        return s != null
+            ? local(s = null, () => M0(s)) // 1
+            : (Action?)null;
+
+        Action local(string? s, Action a1)
+        {
+            return a1;
```

Comment:
Can you add a variation where `s` is assigned null in the local function itself?